Tests/WalletConnectTests/SessionEngineTests.swift

```diff
@@ -56,11 +56,9 @@ final class SessionEngineTests: XCTestCase {
         let agreementKeys = AgreementKeys.stub()
         let topicB = String.generateTopic()
         cryptoMock.setAgreementSecret(agreementKeys, topic: topicB)
-        
         let proposal = SessionProposal.stub(proposerPubKey: AgreementPrivateKey().publicKey.hexRepresentation)
-        
         try? engine.settle(topic: topicB, proposal: proposal, accounts: [], namespaces: [Namespace.stub()])
-        
+        usleep(100)
```

Comment:
Why are these `usleep()` calls needed?

Reply:
settle method has an async Task in it's body and it is not async by itself yet. Assertions are executed synchronously after settle call and concurrently to `networkingInteractor.subscribe()`
I plan to make settle asyn as well in the future so we can remove `usleet()`